library/core/src/option.rs

```diff
@@ -807,8 +840,8 @@ impl<T> Option<T> {
     /// ```
     #[inline]
     #[stable(feature = "option_entry", since = "1.20.0")]
-    pub fn get_or_insert(&mut self, v: T) -> &mut T {
-        self.get_or_insert_with(|| v)
+    pub fn get_or_insert(&mut self, val: T) -> &mut T {
+        self.get_or_insert_with(|| val)
```

Comment:
I think `v` is sufficiently clear and the current options also use `v`, but I don't understand how is `val` different from `v`, there could only be one possible choice here right? If we go for `val` for better IDE preview, why not go for `value` to be clearer, `val` still feels hanging for me, it could mean `valet`? But personally I think `v` is already very clear since I can't think of any other different input choices.
```suggestion
    pub fn get_or_insert(&mut self, v: T) -> &mut T {
        self.get_or_insert_with(|| v)
```

Reply:
IMO the fewer single letter variable names the better, but the standard library is so riddled with variable name inconsistencies that this seems silly to argue about — it might be more productive to try to get people to agree on certain naming guidelines first, and go from there.

Reply:
I think on this case, no one really care ^^, get a clear guideline on this would take a lot of time so just let the author of the pull request decide. This file also contains `value`. @timvermeulen "the fewer single letter variable names the better" I couldn't disagree more. Explicit is better than implicit. A code is way more read than write.

Reply:
@Stargateur I think we're in agreement? I'm saying that I'd like as many variables as possible to not be a single letter.

Reply:
(I think "as possible" is overstated, since I see no issue with closures like `|x| x + 1`.  But I agree in the abstract, and particularly that there's no consistency right now so whatever the PR author did is usually fine.  I'm just glad that parameter names aren't part of the public API -- unlike C# -- so if we eventually do a consistency pass over things it won't be a breaking change.)

Reply:
Yeah, definitely an exaggeration, I also think `i` is a pretty decent variable name for an index (when the scope is sufficiently small).

Reply:
@timvermeulen This patch I suggested was actually what the author originally did, I just thought that was better so I did this comment.